api/core.go

```diff
@@ -143,33 +143,11 @@ func (s *coreService) SubmitTransaction(ctx context.Context, req *protoapi.Submi
 
 	txResult, err := s.blockchain.SubmitTransactionSync(ctx, req.Tx)
 	if err != nil {
-		// This is Tendermint's specific error signature
-		if _, ok := err.(interface {
-			Code() uint32
-			Details() string
-			Error() string
-		}); ok {
-			s.log.Debug("unable to submit transaction", logging.Error(err))
-			fullError := apiError(codes.InvalidArgument, err)
-			if txResult != nil {
-				return &protoapi.SubmitTransactionResponse{
-					Success: false,
-					Code:    txResult.Code,
-					Data:    txResult.Data.String(),
-					Log:     txResult.Log,
-					Height:  0,
-					TxHash:  txResult.Hash.String(),
-				}, fullError
-			}
-			return nil, fullError
-		}
-		s.log.Debug("unable to submit transaction", logging.Error(err))
-
 		return nil, apiError(codes.Internal, err)
 	}
 
 	return &protoapi.SubmitTransactionResponse{
-		Success: true,
+		Success: txResult.Code == 0,
 		Code:    txResult.Code,
 		Data:    txResult.Data.String(),
```

Comment:
This always return the string of the error, e.g: I just tested it, and it returns the hex encoded string: 70617274792068617320696E73756666696369656E7420746F6B656E7320746F207375626D69742064656C65676174696F6E207265717565737420696E20746869732065706F6368
Which is: party has insufficient tokens to submit delegation request in this epoch.

Should we hex decode the txResult.Data field everytime instead?